janus/agents/farmer.py

```diff
@@ -43,3 +52,11 @@ def update_switch(self):
          """
         self.alpha += 0.1
         self.beta -= 0.01
+
+    def encode(self):
+        return json.dumps(self.__dict__)
+
+def decode(s):
```

Comment:
This is a standalone function, to return a Farmer.  It could be implemented as `importFrom(s)` and overwrite an existing Farmer, but I think this is cleaner.